src/transformers/models/character_bert/tokenization_character_bert.py

```diff
@@ -0,0 +1,979 @@
+            character_ids_ = [(index - 1) for index in character_ids]
+            utf8_codes = list(
+                filter(
+                    lambda x: (
+                        (x != self.pad_character_id) and (x != self.bow_character_id) and (x != self.eow_character_id)
+                    ),
+                    character_ids_,
+                )
+            )
+            return bytes(utf8_codes).decode("utf-8")
+
+    def convert_ids_to_tokens(
+        self, ids: Union[int, List[int]], skip_special_tokens: bool = False
+    ) -> Union[str, List[str]]:
+        # NOTE: the first condition needed to be adapted for character_ids instead
+        # of token ids.
+        """
+        Converts a single index or a sequence of indices in a token or a sequence of tokens, using the vocabulary and
+        added tokens.
+
+        Args:
+            ids (`int` or `List[int]`):
+                The token id (or token ids) to convert to tokens.
+            skip_special_tokens (`bool`, *optional*, defaults to `False`):
+                Whether or not to remove special tokens in the decoding.
+
+        Returns:
+            `str` or `List[str]`: The decoded token(s).
+        """
+        if isinstance(ids, list) and isinstance(ids[0], int):
+            # if ids in self._added_tokens_decoder:
+            #    return self._added_tokens_decoder[ids].content
+            # else:
+            return self._convert_id_to_token(ids)
+        tokens = []
+        for character_ids in ids:
+            tokens.append(self._convert_id_to_token(character_ids))
+        return tokens
+
+    def convert_tokens_to_string(self, tokens):
+        """Converts a sequence of tokens (string) in a single string."""
+        out_string = " ".join(tokens).replace(" ##", "").strip()
+        return out_string
+
+    # TODO: check the methods below
+    def build_inputs_with_special_tokens(
+        self, token_ids_0: List[List[int]], token_ids_1: Optional[List[List[int]]] = None
+    ) -> List[List[int]]:
+        """
+        Build model inputs from a sequence or a pair of sequence for sequence classification tasks by concatenating and
+        adding special tokens. A CHARACTER_BERT sequence has the following format:
+
+        - single sequence: `[CLS] X [SEP]`
+        - pair of sequences: `[CLS] A [SEP] B [SEP]`
+
+        Args:
+            token_ids_0 (`List[int]`):
+                List of IDs to which the special tokens will be added.
+            token_ids_1 (`List[int]`, *optional*):
+                Optional second list of IDs for sequence pairs.
+
+        Returns:
+            `List[int]`: List of [input IDs](../glossary#input-ids) with the appropriate special tokens.
+        """
+        if token_ids_1 is None:
+            return [self.cls_token_id] + token_ids_0 + [self.sep_token_id]
+        cls = [self.cls_token_id]
+        sep = [self.sep_token_id]
+        return cls + token_ids_0 + sep + token_ids_1 + sep
+
+    def get_special_tokens_mask(
```

Comment:
we should use copied from here

Reply:
The typing in the method signature changes as well as the docstring. I tried to figure out a way to use Copied from to keep everything the same but only change these type hints but it seems impossible in this case.
I can either:
- use Copied from but have wrong types
- not use Copied from

Reply:
Alright ! Thanks for explaining